WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java

```diff
@@ -463,23 +451,56 @@ public void run() {
     }
 
     private void toggleItemSelected(GridViewHolder holder, int position) {
-        if (mCursor == null || !isValidPosition(position)) {
+        if (!isValidPosition(position)) {
             return;
         }
-        mCursor.moveToPosition(position);
-        int columnIndex = mCursor.getColumnIndex(MediaModelTable.ID);
-        if (columnIndex != -1) {
-            int localMediaId = mCursor.getInt(columnIndex);
-            boolean isSelected = mSelectedItems.contains(localMediaId);
-            setItemSelectedByPosition(holder, position, !isSelected);
-        }
+        int localMediaId = mMediaList.get(position).getId();
+        boolean isSelected = mSelectedItems.contains(localMediaId);
+        setItemSelectedByPosition(holder, position, !isSelected);
     }
 
     public void setSelectedItems(ArrayList<Integer> selectedItems) {
-        mSelectedItems = selectedItems;
+        mSelectedItems.clear();
+        mSelectedItems.addAll(selectedItems);
         if (mCallback != null) {
             mCallback.onAdapterSelectionCountChanged(mSelectedItems.size());
         }
         notifyDataSetChanged();
     }
+
+    /*
+     * returns true if the passed list is the same as the existing one
+     */
+    private boolean isSameList(@NonNull List<MediaModel> mediaList) {
+        if (mediaList.size() != mMediaList.size()) {
+            return false;
+        }
+
+        for (MediaModel media: mediaList) {
+            // make sure this media item is in the existing media list
+            int index = indexOfId(media.getId());
```

Comment:
Isn't better to have a routine `getMediaWithID` that returns either the element or null. So we can just check for null, or check if the state are equals without 2 calls to `get` on `mMediaList`.